test/nullable.jl

```diff
@@ -388,6 +393,119 @@ end
+@test map(x -> x ? 0 : 0.0, Nullable(true))   === Nullable(0)
+@test map(x -> x ? 0 : 0.0, Nullable(false))  === Nullable(0.0)
+@test map(x -> x ? 0 : 0.0, Nullable{Bool}()) |> isnull_oftype(Union{})
+
+# broadcast and elementwise
+@test sin.(Nullable(0.0))            === Nullable(0.0)
+@test sin.(Nullable{Float64}())      |> isnull_oftype(Float64)
+@test @inferred(broadcast(sin, Nullable(0.0)))       === Nullable(0.0)
+@test @inferred(broadcast(sin, Nullable{Float64}())) |> isnull_oftype(Float64)
+
+@test Nullable(8) .+ Nullable(10)     === Nullable(18)
+@test Nullable(8) .- Nullable(10)     === Nullable(-2)
+@test Nullable(8) .+ Nullable{Int}()  |> isnull_oftype(Int)
+@test Nullable{Int}() .- Nullable(10) |> isnull_oftype(Int)
+
+@test log.(10, Nullable(1.0))                 === Nullable(0.0)
+@test log.(10, Nullable{Float64}())           |> isnull_oftype(Float64)
+@test log.(Nullable(10), Nullable(1.0))       === Nullable(0.0)
+@test log.(Nullable(10), Nullable{Float64}()) |> isnull_oftype(Float64)
+
+@test Nullable(2) .^ Nullable(4)      === Nullable(16)
+@test Nullable(2) .^ Nullable{Int}()  |> isnull_oftype(Int)
+
+# multi-arg broadcast
+@test Nullable(1) .+ Nullable(1) .+ Nullable(1) .+ Nullable(1) .+ Nullable(1) .+
+    Nullable(1) === Nullable(6)
+@test Nullable(1) .+ Nullable(1) .+ Nullable(1) .+ Nullable{Int}() .+
+    Nullable(1) .+ Nullable(1) |> isnull_oftype(Int)
+
+# these are not inferrable because there are too many arguments
+us = map(Nullable, 1:20)
+@test broadcast(max, us...) === Nullable(20)
+@test isnull(broadcast(max, us..., Nullable{Int}()))
+
+# test all elementwise operations
+for (eop, op) in ((.+, +), (.-, -), (.*, *), (./, /), (.\, \), (.//, //),
+                  (.==, ==), (.<, <), (.!=, !=), (.<=, <=), (.÷, ÷), (.%, %),
+                  (.<<, <<), (.>>, >>), (.^, ^))
+    # op(1, 1) chosen because it works for all operations
+    res = op(1, 1)
+    @test eop(Nullable(1), Nullable(1))         === Nullable(res)
+    @test eop(Nullable{Int}(), Nullable(1))     |> isnull_oftype(typeof(res))
+    @test eop(Nullable(1), Nullable{Int}())     |> isnull_oftype(typeof(res))
+    @test eop(Nullable{Int}(), Nullable{Int}()) |> isnull_oftype(typeof(res))
+    @test eop(Nullable(1), 1)                   === Nullable(res)
+    @test eop(1, Nullable(1))                   === Nullable(res)
+end
+
+# test reasonable results for Union{}
+# the exact types of these is finnicky and depends on implementation details
+# but is guaranteed to be at worst concrete and possibly Union{} on a good day
+@test isnull(@inferred(Nullable() .+ Nullable()))
+@test isnull(@inferred(Nullable() .+ 1))
+@test isnull(@inferred(Nullable() .+ Nullable(1)))
+
+# test that things don't pessimize because of non-homogenous types
+@test Nullable(10.5) ===
+    @inferred(broadcast(+, 1, 2, Nullable(3), Nullable(4.0), Nullable(1//2)))
+
+# broadcasting for arrays
+@test isequal(@inferred(broadcast(+, [1, 2, 3], Nullable{Int}(1))),
```

Comment:
Doesn't sound too bad to me. I guess you should add heterogeneous types, abstract types, and a few three- and four-argument calls. Maybe tuples too?

And again, would be nice to check that the fast path is taken by replacing `Nullable()` with e.g. `Nullable(1, false)`, and by checking that the nulls in the output array contain the value you expect.